### PR TITLE
#263: Firefox timer recorder: Wrong request method recorded

### PR DESCRIPTION
**Pull request by jowerner**
- determine the correct request method for redirected requests and record that one
- added a test case to create certain redirect scenarios for verification purposes

### CODE REVIEW

File: xlt-timerrecorder/background.js [hunk 1]
```diff
@@ -383,7 +413,7 @@ browser.webRequest.onBeforeRedirect.addListener(function(details) {
       request.responseStart = details.timeStamp;
       request.responseEnd = details.timeStamp;
 
-      request.method = details.method;
+      //request.method = details.method;
```
[the remark this answers is not in the document]
Comment (jowerner) [reply xlt-timerrecorder/background.js:416]: Fixed using `request.method = request.method || details.method;`.

File: xlt-timerrecorder/background.js [hunk 2]
```diff
@@ -407,7 +437,7 @@ browser.webRequest.onResponseStarted.addListener(function(details) {
       const request = getRequestEntry(details.tabId, details.requestId, details.url);
       request.responseStart = details.timeStamp;
 
-      request.method = details.method;
+      //request.method = details.method;
```
[the remark this answers is not in the document]
Comment (jowerner) [reply xlt-timerrecorder/background.js:440]: Fixed using `request.method = request.method || details.method;`.

File: xlt-timerrecorder/background.js [hunk 3]
```diff
@@ -430,7 +460,7 @@ browser.webRequest.onCompleted.addListener(function(details) {
       const request = getRequestEntry(details.tabId, details.requestId, details.url);
       request.responseEnd = details.timeStamp;
 
-      request.method = details.method;
+      //request.method = details.method;
```
[the remark this answers is not in the document]
Comment (jowerner) [reply xlt-timerrecorder/background.js:463]: Fixed using `request.method = request.method || details.method;`.

File: xlt-timerrecorder/background.js [hunk 4]
```diff
@@ -220,6 +220,21 @@ function getRequestEntry(tabId, requestId, url) {
...
+ * 
+ * @return the direct parent request entry, or undefined if not found
+ */
+function getParentRequestEntry(tabId, requestId) {
+  const requestMap = TabRequestsMap[tabId];
+  const urlMap = requestMap && requestMap[requestId] || {};
+  const entries = Object.values(urlMap);
+  const reversedEntries = Array.from(entries).reverse();
+
+  return reversedEntries.find((v) => v.statusCode !== null);
```
[the remark this answers is not in the document]
Comment (jowerner) [reply xlt-timerrecorder/background.js:235]: Fixed using a traditional function.

**Review by h-arlt — APPROVED** [review]

**Pull request merged**